parlai/core/params.py

```diff
@@ -599,8 +597,15 @@ def parse_args(self, args=None, namespace=None, print_args=True):
         # set environment variables
         if self.opt.get('download_path'):
             os.environ['PARLAI_DOWNPATH'] = self.opt['download_path']
+        elif os.environ.get('PARLAI_DOWNPATH') is None:
+            os.environ['PARLAI_DOWNPATH'] = os.path.join(self.parlai_home, 'downloads')
         if self.opt.get('datapath'):
             os.environ['PARLAI_DATAPATH'] = self.opt['datapath']
+        elif os.environ.get('PARLAI_DATAPATH') is None:
+            os.environ['PARLAI_DATAPATH'] = os.path.join(self.parlai_home, 'data')
+
+        self.opt['download_path'] = os.environ['PARLAI_DOWNPATH']
```

Comment:
the self.opt = os.environ bit is surprising but I think your logic is correct. Maybe you can leave a comment saying the priorities. `# --datapath > environ[PARLAI_DATAPATH] > parlai_dir/downloads (default)`.

Reply:
added